editor/src/app/inputs/text-input.component.ts

```diff
@@ -43,6 +43,7 @@ import { TextInputService } from './text-input.service';
       display: block;
     }
   `],
+  changeDetection: ChangeDetectionStrategy.OnPush,
```

Comment:
In short, this disables change detection and makes Angular only update the component if an observable with the async pipe is updated. Basically Angular only uses subscriptions to detect the change.
I guess this works in this scenario because we re-render the component entirely, so it doesn't matter if we detect changes after that.
However, it is not the best approach to avoid the exception. To avoid it normally, you must ensure one-direction data flow.